### PR TITLE
items: complete item model

### PR DESCRIPTION
**Pull request by Garfield-fr**
* Adds new fields on item detail.
* Adds masked holdings and item on document detail view.
* Activated buttons on item detail view.
* Fixes a bug on holdings availability.

Co-Authored-by: Bertrand Zuchuat <bertrand.zuchuat@rero.ch>

## Dependencies

My PR depends on branch:
- https://github.com/rero/rero-ils/tree/US-1906-item-model

## How to test?

- new fields added to the item record
* url
* pac_code
* price
* _masked
* legacy_checkout_count (migrated from Virtua, not possible to edit)
* legacy_circulation_rules (migrated from Virtua, not possible to edit)

- new fields added to the document record
* _masked

- the new fields are displayed in the `item detailed view`
- A new `eye` badge in the admin view to indicate that `item` or `holdings is masked.
- a masked record is not displayed in the public interface
- when a librarian masks all items of a standard holdings,
the system masks the standard holdings automatically.
The standard holdings will be unmasked when there is
at least one unmasked item attached to it.

## Code review check list

- [ ] Commit message template compliance.
- [ ] Commit message without typos.
- [ ] File names.
- [ ] Functions names.
- [ ] Functions docstrings.
- [ ] Unnecessary commited files?
- [ ] Cypress tests successful?

### CODE REVIEW

File: projects/admin/src/app/record/detail-view/item-detail-view/item-detail-view.component.html [hunk 1]
```diff
@@ -24,10 +24,12 @@ <h1>{{ 'Barcode' | translate }} {{ record.metadata.barcode }}</h1>
 
       <!-- INHERITED CALL NUMBER -->
       <ng-container *ngIf="record | itemHoldingsCallNumber | async as callNumber">
-        <dt class="col-sm-3 offset-sm-2 offset-md-0 label-title" translate>Call number </dt>
-        <dd class="col-sm-7 col-md-8 mb-0">
-          <shared-inherited-call-number [item]="record" context="first"></shared-inherited-call-number>
-        </dd>
+        <ng-container *ngIf="(callNumber | json) != ({ first: {}, second: {} } | json)">
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/item-detail-view/item-detail-view.component.html:27]: strange but if it works ! 👍

File: projects/admin/src/app/record/detail-view/item-detail-view/item-detail-view.component.html [hunk 2]
```diff
@@ -104,6 +106,41 @@ <h1>{{ 'Barcode' | translate }} {{ record.metadata.barcode }}</h1>
...
         </span>
       </dd>
+      <!-- PRICE -->
+      <ng-container *ngIf="record.metadata | keyExists:'price'">
+        <dt class="col-sm-3 offset-sm-2 offset-md-0 label-title" translate>
+          Price
+        </dt>
+        <dd class="col-sm-7 col-md-8 mb-0">
+          <!-- TODO: Implement currency with organisation currency symbol -->
+          {{ record.metadata.price }}
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/item-detail-view/item-detail-view.component.html:116]: You can use the `currency` pipe. Already use into circulation module 
```
{{ totalAmountOfFee | currency: organisation.default_currency }}
```

File: projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html [hunk 3]
```diff
@@ -125,6 +125,15 @@ <h3 id="{{'doc-altgr-title-' + i}}">{{ altgr_title }}</h3>
...
       </div>
+
+      <!-- MASKED -->
+      <div class="row">
+        <div class="col">
+          <admin-record-masked [record]="record" [withLabel]="true"></admin-record-masked>
+        </div>
+      </div>
+
+
```
Comment (lauren-d) [inline projects/admin/src/app/record/detail-view/document-detail-view/document-detail-view.component.html:136]: one blank line could be removed

File: projects/admin/src/app/record/detail-view/record-masked/record-masked.component.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  <container-element [ngSwitch]="context">
+    <some-element *ngSwitchCase="'eye'">
+      <ng-container *ngIf="record.metadata | keyExists:'_masked'; else noEyeMasked">
+        <i
+          class="fa"
+          aria-hidden="true"
+          title="{{ (record.metadata._masked ? 'Masked' : 'No masked') | translate }}"
+          [ngClass]="{ 'fa-eye-slash text-danger': record.metadata._masked, 'fa-eye text-success': !record.metadata._masked }"
+        ></i>
+        <label class="ml-1" *ngIf="withLabel">{{ (record.metadata._masked ? 'Masked' : 'No masked') | translate }}</label>
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/record-masked/record-masked.component.ts:31]: Does it OK with string extraction ?

File: projects/admin/src/app/record/detail-view/record-masked/record-masked.component.ts [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+      </ng-container>
+      <ng-template #noEyeMasked>
+        <i class="fa fa-eye text-success" title="No masked" aria-hidden="true"></i>
+        <label class="ml-1" *ngIf="withLabel">{{ 'No masked' | translate }}</label>
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/record-masked/record-masked.component.ts:35]: `transalte` on label tag

File: projects/admin/src/app/record/detail-view/record-masked/record-masked.component.ts [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+          class="fa fa-circle"
+          title="{{ (record.metadata._masked ? 'Masked' : 'No masked') | translate }}"
+          [ngClass]="{'text-danger': !record.metadata._masked, 'text-success': record.metadata._masked }"
+          aria-hidden="true"
+        ></i>
+        <label class="ml-1" *ngIf="withLabel">{{ (record.metadata._masked ? 'Masked' : 'No masked') | translate }}</label>
+      </ng-container>
+      <ng-template #noBulletMasked>
+        <i class="fa fa-circle text-danger mr-1" title="No masked" aria-hidden="true"></i>
+        <label class="ml-1" *ngIf="withLabel">{{ 'No masked' | translate }}</label>
```
Comment (zannkukai) [inline projects/admin/src/app/record/detail-view/record-masked/record-masked.component.ts:50]: same here

**Review by lauren-d — APPROVED** [review]

**Review by zannkukai — CHANGES_REQUESTED** [review]
Some minor changes

**Review by zannkukai — APPROVED** [review]

**Comment by benerken** [comment]
- [ ] If I add an URL to an item, the URL is displayed in the public view (document detailled view) but not in the profesionnal view.
See document 2000556 on ilsdev

**Review by pronguen — COMMENTED** [review]
- [ ] **[important]** The field `_masked` should be displayed only if the resource is masked. Otherwise, the information is not relevant for the librarian.
- [ ] Item detailed view: a masked resource is displayed with a green dot. It would be better to display it without label and with the same red eye. Example: 
image
- [ ] Holdings detailed view:
    - [ ] the "Masked" info should also be displayed for the holdings, just before the call number.
    - [ ] the "Masked" info should also be displayed for the issues, like this for instance:
    image
- [ ] Document detailed view: the "Masked" info should be displayed also for issue items. Example: test1 and test2 are both masked:
image
- [ ] The facet "library" in the organisation view does not take into account the masked resources to display the number of occurrences. *This can be done later if it requires an effort, in that case please open an issue.*

**Review by benerken — APPROVED** [review]

**Pull request merged**